### PR TITLE
Improved out-of-tree cloud provider support

### PR DESCRIPTION
**Pull request by jakolehm**
Allows to integrate out-of-tree kubernetes cloud providers better (in-tree cloud providers are deprecated already in kubernetes).

### CODE REVIEW

**Comment by jnummelin** [comment]
Looks pretty good to me. @jakolehm Are you still planning to refactor this as its marked as draft?

**Comment by jakolehm** [comment]
> Looks pretty good to me. @jakolehm Are you still planning to refactor this as its marked as draft?

I think it's pretty ok already. We can refactor later if needed in a separate pr.

**Comment by jakolehm** [comment]
@jnummelin @kke PTAL

File: lib/pharos/resources/hcloud/hcloud-ccm.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - key: "node.cloudprovider.kubernetes.io/uninitialized"
+        value: "true"
+        effect: "NoSchedule"
+      - key: "CriticalAddonsOnly"
+        operator: "Exists"
+      # cloud controller manages should be able to run on masters
+      - key: "node-role.kubernetes.io/master"
+        effect: NoSchedule
+      containers:
+      - image: hetznercloud/hcloud-cloud-controller-manager:v1.2.0
```
Comment (jnummelin) [inline lib/pharos/resources/hcloud/hcloud-ccm.yml]: Should we sync these images to Pharos CDN too?

**Comment by jakolehm** [comment]
> The only issue/question is whether we should sync the cloud controller images to Pharos CDN.

Maybe not necessary at this point. All of these are "public cloud" where airgap is not usually required.

**Comment by jakolehm** [comment]
@nevalla toleration is now there.

**Pull request merged**